### PR TITLE
Fixed usage of global mock variables in unit-tests

### PR DESCRIPTION
**Pull request by priyanka19-98**
The global mock variables were being reused in may tests, due to which some tests were
failing to run individually, this commit reuses mock variables by creating a DeepCopy of global mock variables to prevent
modification of global mock variables.

There were multiple duplicate functions to create Fake Scheme for the tests,
this commit keeps only one createFakeScheme function for testing StorageCluster controller.

Above two changes are dependent on each other to make tests run individually and as a whole by `make unit-test` . 

Signed off by: Priyanka Jiandani <pjiandan@redhat.com>

### CODE REVIEW

File: controllers/storagecluster/storagecluster_controller_test.go [hunk 1]
```diff
@@ -81,13 +82,13 @@ var mockStorageClusterWithArbiter = &api.StorageCluster{
 
 var mockCephCluster = &rookCephv1.CephCluster{
 	ObjectMeta: metav1.ObjectMeta{
-		Name:      generateNameForCephCluster(mockStorageCluster),
+		Name:      generateNameForCephCluster(mockStorageCluster.DeepCopy()),
```
Comment (umangachapagain) [inline controllers/storagecluster/storagecluster_controller_test.go:85]: Even if you don't use DeepCopy here, it's fine. Because, generate function won't modify the object.

File: controllers/storagecluster/storagecluster_controller_test.go [hunk 2]
```diff
@@ -81,13 +82,13 @@ var mockStorageClusterWithArbiter = &api.StorageCluster{
...
 		Namespace: mockStorageCluster.Namespace,
 	},
 }
 
 var mockCephClusterNamespacedName = types.NamespacedName{
-	Name:      generateNameForCephCluster(mockStorageCluster),
+	Name:      generateNameForCephCluster(mockStorageCluster.DeepCopy()),
```
Comment (umangachapagain) [inline controllers/storagecluster/storagecluster_controller_test.go:91]: Even if you don't use DeepCopy here, it's fine. Because, generate function won't modify the object.

File: controllers/storagecluster/storagecluster_controller_test.go [hunk 3]
```diff
@@ -158,9 +159,9 @@ func getMockDeviceSets(name string, count int, replica int, portable bool) []api
 			Name:                name,
 			Count:               count,
 			Replica:             replica,
-			DataPVCTemplate:     mockDataPVCTemplate,
-			MetadataPVCTemplate: mockMetaDataPVCTemplate,
-			WalPVCTemplate:      mockWalPVCTemplate,
+			DataPVCTemplate:     *mockDataPVCTemplate.DeepCopy(),
```
Comment (umangachapagain) [inline controllers/storagecluster/storagecluster_controller_test.go:162]: Why did you make it a pointer? Is it required to fix unit test?

File: controllers/storagecluster/initialization_reconciler_test.go [hunk 4]
```diff
@@ -163,8 +157,8 @@ func createFakeInitializationStorageClusterReconciler(t *testing.T, obj ...runti
 func createFakeInitializationStorageClusterReconcilerWithPlatform(t *testing.T,
 	platform *Platform,
 	obj ...runtime.Object) StorageClusterReconciler {
-	scheme := createFakeInitializationScheme(t, obj...)
-	obj = append(obj, mockNodeList)
+	scheme := createFakeScheme(t)
```
Comment (umangachapagain) [inline controllers/storagecluster/initialization_reconciler_test.go:160]: Earlier this scheme had a lot of api schemes added. Is it still the case or it's just the default scheme?

File: controllers/storagecluster/storagecluster_controller_test.go (hunk 1 above)
Comment (priyanka19-98) [reply controllers/storagecluster/storagecluster_controller_test.go:85]: yes, true. I wanted to make sure that we don't use mock variables directly regardless of their modification.

File: controllers/storagecluster/storagecluster_controller_test.go (hunk 3 above)
Comment (priyanka19-98) [reply controllers/storagecluster/storagecluster_controller_test.go:162]: The DeepCopy() returns a pointer, but we don't need a pointer, hence we use the pointer value by `*`

File: controllers/storagecluster/initialization_reconciler_test.go (hunk 4 above)
Comment (priyanka19-98) [reply controllers/storagecluster/initialization_reconciler_test.go:160]: yes, this scheme has a lot of api schemes.  `createFakeInitializationScheme()` also had a lot of api schemes added, which was duplicated.

**Review by umangachapagain — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -119,7 +119,7 @@ make unit-test
 
 *It is of special note that many of the mock objects in the [StorageCluster
 tests](./pkg/controller/storagecluster/storagecluster_controller_test.go) are
-reused in many other test cases. They may be useful in developing your own.*
+reused in many other test cases. Please avoid using the global mock variables directly, instead use the mock variables by creating DeepCopy() to prevent modification of global mock variables. They may be useful in developing your own.*
```
Comment (jarrpa) [inline CONTRIBUTING.md:122]: This line should be properly wrapped as all the other text in the markdown. But that can be a separate PR, I'm not about to hold the code changes for this at this stage.
